index.html

```diff
@@ -195,69 +335,537 @@ <h2>
-          different values (or values in a different order).
+          <p>
+            Array of button states for all buttons of the gamepad. It is
+            RECOMMENDED that buttons appear in decreasing importance such that
+            the primary button, secondary button, tertiary button, and so on
+            appear as elements 0, 1, 2, ... in the buttons array. The same
+            object MUST be returned until the [=user agent=] needs to return
+            different values (or values in a different order).
+          </p>
+          <p>
+            The {{Gamepad/buttons}} getter steps are:
+          </p>
+          <ol>
+            <li>Return [=this=].{{Gamepad/[[buttons]]}}.
+            </li>
+          </ol>
         </dd>
       </dl>
+      <section>
+        <h3>
+          Receiving inputs
+        </h3>
+        <p>
+          When the system <dfn>receives new button or axis input values</dfn>,
+          run the following steps:
+        </p>
+        <ol>
+          <li>Let |gamepad:Gamepad| be the {{Gamepad}} object representing the
+          device that received new button or axis input values.
+          </li>
+          <li>[=Queue a global task=] on the [=relevant global object=] of
+          |gamepad| using the [=gamepad task source=] to [=update gamepad
+          state=] for |gamepad|.
+          </li>
+        </ol>
+        <p>
+          To <dfn>update gamepad state</dfn> for |gamepad:Gamepad|, run the
+          following steps:
+        </p>
+        <ol>
+          <li>Let |now:DOMHighResTimeStamp| be the [=current high resolution
+          time=].
+          </li>
+          <li>Set |gamepad|.{{Gamepad/[[timestamp]]}} to |now|.
+          </li>
+          <li>Run the steps to [=map and normalize axes=] for |gamepad|.
+          </li>
+          <li>Run the steps to [=map and normalize buttons=] for |gamepad|.
+          </li>
+          <li>Let |navigator:Navigator| be |gamepad|'s [=relevant global
+          object=]'s {{Navigator}} object.
+          </li>
+          <li>If |navigator|.{{Navigator/[[hasGamepadGesture]]}} is `false` and
+          |gamepad| [=contains a gamepad user gesture=]:
+            <ol>
+              <li>Set |navigator|.{{Navigator/[[hasGamepadGesture]]}} to
+              `true`.
+              </li>
+              <li>[=list/For each=] |connectedGamepad:Gamepad?| of
+              |navigator|.{{Navigator/[[gamepads]]}}:
+                <ol>
+                  <li>If |connectedGamepad| is not equal to `null`:
+                    <ol>
+                      <li>Set |connectedGamepad|.{{Gamepad/[[exposed]]}} to
+                      `true`.
+                      </li>
+                      <li>Set |connectedGamepad|.{{Gamepad/[[timestamp]]}} to
+                      |now|.
+                      </li>
+                      <li>If |gamepad|'s [=relevant global object=] is a
+                      {{Window}} object, [=queue a global task=] on |gamepad|'s
```

Comment:
Here, we might want to also do the fully active check (as we are checking the {{Window}} is still there, per #149).... we can come back to this tho.

Reply:
I've added a fully active check so we won't try to fire events for non-fully-active Documents.

I removed the Window check since I think we'll implicitly make that check when trying to get the associated (or responsible) Document. If the global object isn't a Window then it won't have a Document. Is that correct?